projects/igniteui-angular/src/lib/drop-down/drop-down.component.ts

```diff
@@ -542,17 +542,23 @@ export class IgxDropDownComponent extends IgxDropDownBaseDirective implements ID
         this.onSelection.emit(args);
 
         if (!args.cancel) {
-            this.selection.set(this.id, new Set([newSelection]));
-            if (!this.virtDir) {
-                if (oldSelection) {
-                    oldSelection.selected = false;
+            if (args.newSelection === null ||
+                (args.newSelection.hasOwnProperty('value') && args.newSelection.hasOwnProperty('index') && this.virtDir) ||
+                (args.newSelection instanceof IgxDropDownItemComponent && !args.newSelection.isHeader)) {
+                this.selection.set(this.id, new Set([args.newSelection]));
+                if (!this.virtDir) {
+                    if (oldSelection) {
+                        oldSelection.selected = false;
+                    }
+                    if (newSelection) {
```

Comment:
you checked this on line 529

Reply:
Thanks for pointing this out! This should be `args.newSelection` - we shouldn't try to set `args.newSelection.selected`, if `null` was passed in the selection args.